source/MRMesh/MRFastWindingNumber.cpp

```diff
@@ -179,6 +179,12 @@ VoidOrErrStr FastWindingNumber::calcFromGrid( std::vector<float>& res, const Vec
     return {};
 }
 
+float FastWindingNumber::calcWithDistances( const Vector3f& p, float beta, float maxDistSq, float minDistSq )
+{
+    const auto sign = calc( p, beta ) > 0.5f ? -1.f : +1.f;
+    return sign * std::sqrt( findProjection( p, mesh_, maxDistSq, nullptr, minDistSq ).distSq );
```

Comment:
For best performance, it is better to find unsigned distance first, and only if it is within [min/max] compute the sign